src/mongo/integration/standalone/dbclient_test.cpp

```diff
@@ -240,6 +240,23 @@ TEST_F(DBClientTest, FindAndModifyDuplicateKeyError) {
         OperationException);
 }
 
+TEST_F(DBClientTest, FindAndModifyWriteConcern) {
+    c->insert(TEST_NS, BSON("_id" << 1 << "i" << 1));
+
+    BSONObj result = c->findAndModify(TEST_NS,
+                                      BSON("i" << 1),
+                                      BSON("$inc" << BSON("i" << 1)),
+                                      false,
+                                      false,
+                                      BSONObj(),
+                                      BSONObj(),
+                                      &WriteConcern::journaled);
+
+    ASSERT_EQUALS(result.getIntField("_id"), 1);
+    ASSERT_EQUALS(result.getIntField("i"), 1);
+    ASSERT_EQUALS(c->count(TEST_NS), 1U);
```

Comment:
is it worth doing the whole song & dance with the system profiler to check that the system actually registered the proper write concern?